quoridorcore/src/main/java/com/harush/zitoon/quoridor/core/model/PawnLogic.java

```diff
@@ -87,7 +84,7 @@ public void setInitialCoordinate(Coordinate initialCoordinate) {
     public void setCurrentCoordinate(Coordinate currentCoordinate) {
         int newX = currentCoordinate.getX();
         int newY = currentCoordinate.getY();
-        board.movePawn(currentX, currentY, newX, newY);
+        board.movePawn(currentX, currentY, newX, newY); //we recieve logicResult and doing nothing with it
```

Comment:
You are correct. This should probably be removed from board since board should be less smart, and the PawnLogic is the one that knows if the pawn can move or not. It stayed there as legacy old code which I didn't have time to refactor properly yet.